### PR TITLE
refactor: add metadata basemodel

### PR DESCRIPTION
**Pull request by MartinBernstorff**
For initial feedback. Note that the first commit is a large refactor, so recommend excluding it from the diff when reviewing, e.g. only looking at:

https://github.com/embeddings-benchmark/mteb/pull/260/commits/df7672d0c5259502137d3d038f3deb995279b81e

If we are agreed on the schema, we can merge, and then I'll then start updating the 100+ existing tasks. What do you think @KennethEnevoldsen?

Todo:
- [x] Check that all files with _LANGS and _EVAL_SPLIT use them in their TaskMetadata

### CODE REVIEW

**Review by KennethEnevoldsen — COMMENTED** [review]
A few recommendations otherwise I believe it looks good.

Can you also bump the version number. An example where you transform DDisco would also be great (then the other reviewers can see how it would look in practice).

**Comment by MartinBernstorff** [comment]
> Can you also bump the version number. 
Ah, fair point. Should we perhaps set up semantic versioning?

> An example where you transform DDisco would also be great (then the other reviewers can see how it would look in practice).

Should already be in the commit history?

https://github.com/embeddings-benchmark/mteb/pull/260/commits/df7672d0c5259502137d3d038f3deb995279b81e#diff-35934ee4d3e5d7d6659d8cc6c67991c680adf28f1d1d78ab9701f6978e0dd90b

I'll take a look at the rest 👍

File: mteb/abstasks/TaskMetadata.py
Comment (MartinBernstorff) [reply mteb/abstasks/TaskMetadata.py:26]: Super! Will wait to see if the other reviewers agree, then move it

**Comment by KennethEnevoldsen** [comment]
> Ah, good point. Should we perhaps set up semantic versioning?

#261 

> Should already be in the commit history?

Ah thanks must have missed it.

**Comment by MartinBernstorff** [comment]
Questions before merge:
* The amazon polarity dataset is hosted by MTEB. I have added `https://huggingface.co/datasets/amazon_polarity` as reference, is that correct?

**Comment by KennethEnevoldsen** [comment]
> The amazon polarity dataset is hosted by MTEB. I have added https://huggingface.co/datasets/amazon_polarity as reference, is that correct?

I believe so @Muennighoff will have to confirm though.

**Review by KennethEnevoldsen — APPROVED** [review]

**Comment by MartinBernstorff** [comment]
Should be ready to merge @KennethEnevoldsen 👍

**Comment by KennethEnevoldsen** [comment]
The tests seem to fail @MartinBernstorff it seems to be due to a missing pydantic dependency.

**Comment by MartinBernstorff** [comment]
@KennethEnevoldsen Seems I haven't been added as a collaborator, so I can't merge this. Ready to merge now 👍

**Pull request merged**
